lib/commands/post-install.ts

```diff
@@ -0,0 +1,91 @@
+import {PostInstallCommand} from "../common/commands/post-install";
+let emailValidatorModule = require('email-validator');
+let queryString = require('querystring');
+import * as helpers from "../common/helpers";
+
+export class PostInstallCliCommand extends PostInstallCommand {
+
+    private fileSystem: IFileSystem;
+    private logger: ILogger;
+
+    constructor($fs: IFileSystem,
+        private $httpClient: Server.IHttpClient,
+        private $prompter: IPrompter,
+        private $userSettingsService: IUserSettingsService,
+		$staticConfig: Config.IStaticConfig,
+		$commandsService: ICommandsService,
+		$htmlHelpService: IHtmlHelpService,
+		$options: ICommonOptions,
+		$doctorService: IDoctorService,
+		$analyticsService: IAnalyticsService,
+		$logger: ILogger) {
+            super($fs, $staticConfig, $commandsService, $htmlHelpService, $options, $doctorService, $analyticsService, $logger);
+            this.fileSystem = $fs;
+            this.logger = $logger;
+	}
+
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+            super.execute(args).wait();
+
+           	if (this.shouldAskForEmail()) {
+                this.logger.out("Leave your e-mail address here to subscribe for NativeScript newsletter and product updates, tips and tricks:");
+ 				let email = this.getEmail("(press Enter for blank)").wait();
+ 				this.sendEmail(email);
+ 				this.$userSettingsService.saveSetting("EMAIL_REGISTERED", true).wait();
```

Comment:
if the sendEmail command fails, next time when user installs NS CLI, he'll be prompted again (as the EMAIL_REGISTERED value will never be written in the settings file), is this expected? In case not, you can swap the two lines above.